### PR TITLE
feat: API for project metrics range [WEB-1000]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Create endpoint `/projects/{id}/experiments/metric-ranges`

Example return 
```
{
    "ranges": [
        {
            "metricsName": "validation.validation_error",
            "min": 0.0423911582752162,
            "max": 0.0423911582752162
        },
        {
            "metricsName": "validation.accuracy",
            "min": 0.983180732484076,
            "max": 0.984175955414013
        },
        {
            "metricsName": "validation.validation_loss",
            "min": 0.0496316553445427,
            "max": 0.0521038854505813
        },
        {
            "metricsName": "training.loss",
            "min": 0.0423911582752162,
            "max": 0.0866028890013695
        }
    ]
}
```

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1000

### CODE REVIEW

File: master/internal/api_project.go [hunk 1]
```diff
@@ -316,6 +332,121 @@ func (a *apiServer) GetProjectColumns(
 	return a.getProjectColumnsByID(ctx, req.Id, *curUser)
 }
 
+func (a *apiServer) GetProjectMetricsRange(
```
[the remark this answers is not in the document]
Comment (gt2345) [reply master/internal/api_project.go:336]: Of course

File: master/internal/api_project.go [hunk 2]
```diff
@@ -317,6 +334,106 @@ func (a *apiServer) GetProjectColumns(
...
+	}
+
+	var res []struct {
+		SmallerIsBetter   bool
+		ValidationMetrics *map[string]metrics
+		AvgMetrics        *map[string]metrics
+	}
+
+	if err := query.Scan(ctx, &res); err != nil {
+		return nil, nil, err
```
Comment (NicholasBlaskey) [inline master/internal/api_project.go:396]: wrap this error

File: master/internal/api_project.go [hunk 3]
```diff
@@ -243,19 +266,13 @@ func (a *apiServer) getProjectColumnsByID(
 	if err != nil {
 		return nil, err
```
Comment (NicholasBlaskey) [inline master/internal/api_project.go:267]: wrap this error

File: proto/src/determined/api/v1/api.proto [hunk 4]
```diff
@@ -1834,9 +1834,20 @@ service Determined {
       get: "/api/v1/projects/{id}/columns"
     };
     option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
-      tags: "Internal"
+      tags: [ "Projects", "Internal" ]
     };
   }
+  // Get metrics range for a project.
+  rpc GetProjectMetricsRange(GetProjectMetricsRangeRequest)
```
Comment (NicholasBlaskey) [inline proto/src/determined/api/v1/api.proto:1841]: Maybe name this `GetProjectNumericMetricsRange`

to be explicit here that we aren't returning non numeric metrics

**Review by NicholasBlaskey — APPROVED** [review]
LGTM 

Nice work

**Pull request merged**